presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -290,6 +290,11 @@ public void addTypes(IdentityHashMap<Expression, Type> types)
         this.types.putAll(types);
     }
 
+    public void replaceType(Expression expression, Type type)
```

Comment:
I got a problem here, I added a test in my local code,
assertQuery(
                "SELECT * FROM (VALUES (1.0)) x (a) JOIN (VALUES (3)) y (a) ON y.a > x.a",
                "VALUES (1.0, 3)"
        );
I remove replaceType() in my code and it passed the test, but when I run it in my cluster, this SQL throws exception.